src/WorldFacade.js

```diff
@@ -1,733 +1,809 @@
-
-		this.config = newConfig
-		// pass updates to DiceWorld
-		this.#DiceWorld.updateConfig(newConfig)
-		// pass updates to PhysicsWorld
-		this.#DicePhysics.postMessage({
-			action: 'updateConfig',
-			options: newConfig
-		})
-
-		// make this method chainable
-		return this
-	}
-
-	clear() {
-		// reset indexes
-		this.#collectionIndex = 0
-		this.#groupIndex = 0
-		this.#rollIndex = 0
-		this.#idIndex = 0
-		// reset internal data objects
-		this.rollCollectionData = {}
-		this.rollGroupData = {}
-		this.rollDiceData = {}
-		// clear all rendered die bodies
-		this.#DiceWorld.clear()
+  clear() {
+    // reset indexes
+    this.#collectionIndex = 0;
+    this.#groupIndex = 0;
+    this.#rollIndex = 0;
+    this.#idIndex = 0;
+    // reset internal data objects
+    this.rollCollectionData = {};
+    this.rollGroupData = {};
+    this.rollDiceData = {};
+    // clear all rendered die bodies
+    this.#DiceWorld.clear();
     // clear all physics die bodies
-    this.#DicePhysics.postMessage({action: "clearDice"})
+    this.#DicePhysics.postMessage({ action: "clearDice" });
 
-		// make this method chainable
-		return this
+    // make this method chainable
+    return this;
   }
 
-	hide() {
-		this.canvas.style.display = 'none'
-
-		// make this method chainable
-		return this
-	}
-
-	show() {
-		this.canvas.style.display = 'block'
-
-		// make this method chainable
-		return this
-	}
-
-	// TODO: pass data with roll - such as roll name. Passed back at the end in the results
-	roll(notation, {theme, themeColor, newStartPoint = true} = {}) {
-		// note: to add to a roll on screen use .add method
-		// reset the offscreen worker and physics worker with each new roll
-		this.clear()
-		const collectionId = this.#collectionIndex++
-		
-		this.rollCollectionData[collectionId] = new Collection({
-			id: collectionId,
-			notation,
-			theme,
-			themeColor,
-			newStartPoint
-		})
-
-		const parsedNotation = this.createNotationArray(notation)
-		this.#makeRoll(parsedNotation, collectionId)
-
-		// returns a Promise that is resolved in onRollComplete
-		return this.rollCollectionData[collectionId].promise
-	}
-
-  add(notation, {theme, themeColor, newStartPoint = true} = {}) {
-
-		const collectionId = this.#collectionIndex++
-
-		this.rollCollectionData[collectionId] = new Collection({
-			id: collectionId,
-			notation,
-			theme,
-			themeColor,
-			newStartPoint
-		})
-		
-		const parsedNotation = this.createNotationArray(notation)
-		this.#makeRoll(parsedNotation, collectionId)
-
-		// returns a Promise that is resolved in onRollComplete
-		return this.rollCollectionData[collectionId].promise
+  hide() {
+    this.canvas.style.display = "none";
+
+    // make this method chainable
+    return this;
+  }
+
+  show() {
+    this.canvas.style.display = "block";
+    this.resizeWorld({
+      height: this.canvas.clientHeight,
+      width: this.canvas.clientWidth,
+    });
```

Comment:
In our app we init the dicebox then immediately hide the canvas. We only want the canvas active when dice are being rolled. The problem was issue #54 would happen. This ensure we have a size again when the browser is reset to "block".